Odata-docs/client/grouping-and-aggregation.md

```diff
@@ -0,0 +1,481 @@
+    Amount with min as MinAmount,Amount with max as MaxAmount,
+    $count as NumberOfProductsSoldByCountry,
+    Product/Name with countdistinct as DistinctNumberOfProductsSoldByCountry))
+    ```
+
+10. Member initialization in the key selector
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.GroupBy(
+        d1 => new GroupingKey
+        {
+            CategoryId = d1.Product.CategoryId,
+            Country = d1.Customer.Country
+        },
+	    (d2, d3) => new
+	    {
+	        d2.CategoryId,
+	        d2.Country,
+	        AverageAmount = d3.Average(d4 => d4.Amount),
+	        SumAmount = d3.Sum(d4 => d4.Amount)	   
+	    });
+    ```
+    where `GroupingKey` is a class defined as follows:
+
+    ```csharp
+    class GroupingKey
+    {
+        public string CategoryId { get; set; }
+        public string Country { get; set; }
+    }
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=groupby((Product/CategoryId,Customer/Country),
+    aggregate(Amount with average as AverageAmount,Amount with sum as SumAmount))
+    ```
+
+11. Mixed constructor and member initialization in the result selector
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.GroupBy(
+        d1 => new { d1.Time.Year, CategoryName = d1.Product.Category.Name },
+        (d2, d3) = new GroupedResult(d2.Year, d3.Average(d4 => d4.Amount))
+        {
+            CategoryName = d2.CategoryName,
+            SumAmount = d3.Sum(d4 => d4.Amount)
+        });
+    ```
+    where `GroupedResult` is a class defined as:
+    ```csharp
+    class GroupedResult
+    {
+        public int Year { get; }
+        public string CategoryName { get; set; }
+        public decimal AverageAmount { get; }
+        public decimal SumAmount { get; set; }
+    }
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=groupby((Time/Year,Product/Category/Name),
+    aggregate(Amount with average as AverageAmount,Amount with sum as SumAmount)
+    ```
+
+12. Support for known primitive types' member access and method calls
+
+    _Expression:_
+    ```csharp
+    dataServiceContext.Sales.GroupBy(
+        d1 => new
+        {
+            d1.Time.Year,
+            CategoryName = d1.Product.Category.Name,
+            d1.CurrencyCode
+        },
+        (d2, d3) => new
+        {
+            FoobarLength = "foobar".Length,
+            TenStr = 10.ToString(),
+            YearStr = d2.Year.ToString(),
+            CategoryNameLength = d2.CategoryName.Length,
+            d2.CurrencyCode,
+            AverageAmount = d3.Average(d4 => d4.Amount).ToString(),
+            SumAmount = d3.Sum(d4 => d4.Amount),
+            MinAmount = d3.Min(d4 => d4.Amount).ToString()
+        });
+    ```
+    _Translation:_
+    ```
+    /Sales?$apply=groupby((Time/Year,Product/Category/Name,CurrencyCode),
+    aggregate(Amount with average as AverageAmount,
+    Amount with sum as SumAmount,Amount with min as MinAmount)
+    ```
+
+13. Filter before grouping
```

Comment:
Is filtering after the `GroupBy` supported? What would the result be?

Reply:
@habbes It's supported in the spec but we're not supporting in this release. I'm also not 100% we support it in OData WebApi